src/app/services/hyp3/hyp3-api.service.ts

```diff
@@ -157,6 +167,176 @@ export class Hyp3ApiService {
     );
   }
 
+  public updateJobName$(
+    jobId: string,
+    newProjectName: string,
+  ): Observable<models.Hyp3Job> {
+    const url = `${this.apiUrl}/jobs/${jobId}`;
+
+    if (!newProjectName) {
+      newProjectName = null;
+    }
+
+    return this.http
+      .patch<models.Hyp3Job>(
+        url,
+        { name: newProjectName },
+        { withCredentials: true },
+      )
+      .pipe(map((resp) => resp as models.Hyp3Job));
+  }
+
+  public updateJobsName$(
+    products: models.CMRProduct[],
+    newProjectName: string,
+  ): Observable<{ success: number; failed: number }> {
+    const url = `${this.apiUrl}/jobs`;
+    const jobIds = products.map((product) => product.metadata.job.job_id);
+
+    if (!newProjectName) {
+      newProjectName = null;
+    }
+
+    return from(jobIds).pipe(
+      bufferCount(100),
+      mergeMap((jobIdsBatch) => {
+        return this.http
+          .patch<models.Hyp3Job>(
+            url,
+            { name: newProjectName, job_ids: jobIdsBatch },
+            { withCredentials: true },
+          )
+          .pipe(
+            map(() => ({ success: jobIdsBatch.length, failed: 0 })),
+            catchError(() => of({ success: 0, failed: jobIdsBatch.length })),
+          );
+      }, 3),
+      toArray(),
+      map((results) =>
+        results.reduce(
+          (acc, curr) => ({
+            success: acc.success + curr.success,
+            failed: acc.failed + curr.failed,
+          }),
+          { success: 0, failed: 0 },
+        ),
+      ),
+    );
+  }
+
+  /**
+   * Progress info emitted during rename operations.
+   */
+  public static readonly RenameProgressInfo = class {
+    percent: number;
+    estimatedSecondsRemaining: number | null;
+  };
+
+  /**
+   * Updates job names with progress reporting.
+   * Returns an object with:
+   * - progress$: Observable that emits progress info (percent and estimated time) as batches complete
+   * - result$: Observable that emits the final result when all batches are done
+   */
+  public updateJobsNameWithProgress$(
+    products: models.CMRProduct[],
+    newProjectName: string,
+  ): {
+    progress$: Observable<{
+      percent: number;
+      estimatedSecondsRemaining: number | null;
+    }>;
+    result$: Observable<{
+      success: number;
+      failed: number;
+      failedProjectNames: string[];
+    }>;
+  } {
```

Comment:
I would pull this return type out and give it a name. Similar to `ProjectNameDialogData` and `ProjectNameDialogResult`